sstables/generation_type.hh

```diff
@@ -31,6 +31,17 @@ public:
 
     constexpr bool operator==(const generation_type& other) const noexcept { return _value == other._value; }
     constexpr std::strong_ordering operator<=>(const generation_type& other) const noexcept { return _value <=> other._value; }
+
+    generation_type& operator++() noexcept {
+        ++_value;
+        return *this;
+    }
+
+    generation_type operator++(int) noexcept {
```

Comment:
makes sense to be. as the random uuid might not be completely random. in the case of uuid v1, we might need to the increment the "clock_seq" bits to differentiate the new one from the latest one. so we will have to track the latest generation somewhere. put in other way, the generator of generation could be stateful.

Reply:
What do you mean by not be completely random?
`clock_seq_and_node` is based on the hw rng and unique shard id so it should be globally unique,
then get_time_UUID -> create_time_safe ensures uniqueness of the timestamp part.

Reply:
i see. if we use `UUID_gen` for generating the time uuid, the uniqueness is guaranteed. as it tracks the `_last_used_time`. so the only reason to have `operator++` is just to encapsulate the difference of these two implementations, even the latter does not really depends on the state tracked by the "last generation" -- it has a thread local `UUID_gen::_instance` for this purpose.

Reply:
> What do you mean by not be completely random? `clock_seq_and_node` is based on the hw rng and unique shard id so it should be globally unique, then get_time_UUID -> create_time_safe ensures uniqueness of the timestamp part.

just a small correction: in clock_seq_and_node, the clock seq is a result of pnrng (with rand_r, therefore not hw rng) using time since epoch as its seed. the node is the MAC address, tampered with shard id.